### PR TITLE
Add an impl Debug for HandledNode

### PR DESCRIPTION
**Pull request by jamesray1**

### CODE REVIEW

File: core/src/nodes/handled_node.rs [hunk 1]
```diff
@@ -160,6 +159,22 @@ where
     is_shutting_down: bool
 }
 
+impl<TMuxer, THandler> fmt::Debug for HandledNode<TMuxer, THandler>
+where
+    TMuxer: StreamMuxer,
+    THandler: NodeHandler<Substream = Substream<TMuxer>>,
+    THandler: fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
```
Comment (twittner) [inline core/src/nodes/handled_node.rs]: How about using `fmt::Result`?

File: core/src/nodes/handled_node.rs [hunk 2]
```diff
@@ -160,6 +159,22 @@ where
     is_shutting_down: bool
 }
 
+impl<TMuxer, THandler> fmt::Debug for HandledNode<TMuxer, THandler>
+where
+    TMuxer: StreamMuxer,
+    THandler: NodeHandler<Substream = Substream<TMuxer>>,
+    THandler: fmt::Debug,
```
Comment (twittner) [inline core/src/nodes/handled_node.rs]: How about `THandler: NodeHandler<Substream = Substream<TMuxer>> + fmt::Debug`?

File: core/src/nodes/handled_node.rs (hunk 1 above)
Comment (tomaka) [reply core/src/nodes/handled_node.rs]: Let's use the new suggestion API?

```suggestion
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result<()> {
```

File: core/src/nodes/handled_node.rs [hunk 3]
```diff
@@ -160,6 +159,22 @@ where
     is_shutting_down: bool
 }
 
+impl<TMuxer, THandler> fmt::Debug for HandledNode<TMuxer, THandler>
+where
+    TMuxer: StreamMuxer,
+    THandler: NodeHandler<Substream = Substream<TMuxer>>,
```
Comment (tomaka) [inline core/src/nodes/handled_node.rs]:
```suggestion
    THandler: NodeHandler<Substream = Substream<TMuxer>> + fmt::Debug,
```

File: core/src/nodes/handled_node.rs (hunk 2 above)
Comment (tomaka) [reply core/src/nodes/handled_node.rs]:
```suggestion
```

File: core/src/nodes/handled_node.rs [hunk 4]
```diff
@@ -160,6 +159,21 @@ where
     is_shutting_down: bool
 }
 
+impl<TMuxer, THandler> fmt::Debug for HandledNode<TMuxer, THandler>
+where
+    TMuxer: StreamMuxer,
+    THandler: NodeHandler<Substream = Substream<TMuxer>> + fmt::Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result<()> {
```
Comment (tomaka) [inline core/src/nodes/handled_node.rs]: Oh well, it looks like `fmt::Result` takes no parameter.

```suggestion
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
```

Comment (jamesray1) [reply core/src/nodes/handled_node.rs]: Should've checked myself. ;P

**Pull request merged**